.github/actions/setup-python-matrix/action.yml

```diff
@@ -47,4 +47,4 @@ runs:
       shell: bash
       run: |
         python3.10 -m pip install -U pip
-        python3.10 -m pip install -U wheel setuptools 'tox<4' virtualenv!=20.0.24
+        python3.10 -m pip install -U wheel setuptools 'tox<4' 'virtualenv<20.21.1'
```

Comment:
It looks like v20.21.1 still supported Python 2.7 (https://github.com/pypa/virtualenv/blob/20.21.1/.github/workflows/check.yml#L66) so we can update this to be virtualenv<[20.22.0](https://github.com/pypa/virtualenv/releases/tag/20.22.0).